src/Host/Broker/Impl/Startup/CommonStartup.cs

```diff
@@ -37,34 +38,65 @@ internal class CommonStartup {
 
         internal static void Start(IConfigurationRoot configuration, bool isService) {
             IsService = isService;
-            Configuration = configuration;
-            Configuration.GetSection("startup").Bind(_startupOptions);
-            Configuration.GetSection("security").Bind(_securityOptions);
-            Configuration.GetSection("logging").Bind(_loggingOptions);
-
-            _loggerFactory
-                .AddDebug()
-                .AddConsole(LogLevel.Trace)
-                .AddProvider(new FileLoggerProvider(_startupOptions.Name, _loggingOptions.LogFolder));
-
             if (isService) {
-                _loggerFactory.AddProvider(new EventLogLoggerProvider(LogLevel.Warning, Resources.Text_ServiceName));
+                StartService(configuration);
+            } else {
+                StartApp(configuration);
             }
-            
-            _logger = _loggerFactory.CreateLogger<Program>();
+        }
 
-            if (_startupOptions.Name != null) {
-                _logger.LogInformation(Resources.Info_BrokerName, _startupOptions.Name);
+        private static void ApplyCommonConfiguration(IConfigurationRoot configuration, EventLogLoggerProvider eventLogProvider = null) {
+            // This is needed because we are trying to read the config file. At this point we don't have the config info for the logger 
+            // we generally use in the broker. Log will be visible if the user runs broker interactively for broker running locally. Log 
+            // will be added to windows event log for the remote broker.
+            using (ILoggerFactory configLoggerFactory = new LoggerFactory()) {
+                configLoggerFactory
+                    .AddDebug()
+                    .AddConsole(LogLevel.Trace);
+
+                if (eventLogProvider != null) {
+                    configLoggerFactory.AddProvider(eventLogProvider);
+                }
+
+                ILogger logger = configLoggerFactory.CreateLogger<BrokerService>();
+                try {
+                    string configFile = configuration["config"];
+                    if (configFile != null) {
+                        var configBuilder = new ConfigurationBuilder().AddJsonFile(configFile, optional: false);
+                        configuration = configBuilder.Build();
+                    }
+
+                    Configuration = configuration;
+                    Configuration.GetSection("startup").Bind(_startupOptions);
+                    Configuration.GetSection("security").Bind(_securityOptions);
+                    Configuration.GetSection("logging").Bind(_loggingOptions);
+                } catch (Exception ex) when (!ex.IsCriticalException()) {
```

Comment:
I think this might also throw if command line arguments are messed up, so you might want to change the error message to be more generic to reflect that.